packages/babel-register/src/node.js

```diff
@@ -67,7 +70,7 @@ function compile(code, filename) {
       // Do not process config files since has already been done with the OptionManager
       // calls above and would introduce duplicates.
       babelrc: false,
-      sourceMaps: "both",
+      sourceMaps: "sourceMaps" in opts ? opts.sourceMaps : "both",
```

Comment:
Oh one last comment, sorry. It's technically fine to have the key but have it set to `undefined`, so I'd probably vote for `opts.sourceMaps === undefined ? "both" : opts.sourceMaps`.

Reply:
This might cause some confusion, as it might be the user's intention to pass `undefined` as `falsy`, and this way we are not allowing that.

Reply:
It's technically possible, but at the end of the day that's not what it means in our options processing. `undefined` means non-existant, so for instance if you have `sourceMaps: true` in your `.babelrc`, passing `sourceMaps: false` in the programmatic options would override `true` to `false`, whereas `sourceMaps: undefined` would have no effect.

Reply:
Right, making the change.